### PR TITLE
[6.x/7.x-Prepatch] Scripts/Spells: Fix Shaman Talent Ancestral Guidance

### PR DESCRIPTION
**Pull request by P-Kito**
**Changes proposed**:
- Fix http://www.wowhead.com/spell=108281/ancestral-guidance

**Target branch(es)**: 6.x/7.x-Prepatch

**Tests performed**: Tested ingame, tested build - DONE

### CODE REVIEW

**Comment by P-Kito** [comment]
Working ingame - Ready for merge.

File: src/server/scripts/Spells/spell_shaman.cpp [hunk 1]
```diff
@@ -82,6 +84,85 @@ enum MiscSpells
...
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE))
+                    return false;
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL))
+                    return false;
+                return true;
+            }
+
+            void ResizeTargets(std::list<WorldObject*>& targets)
+            {
+                if (targets.size() > 3)
```
Comment (Shauren) [inline src/server/scripts/Spells/spell_shaman.cpp]: use Trinity::Containers::RandomResizeList (size check before calling it not needed)

File: src/server/scripts/Spells/spell_shaman.cpp [hunk 2]
```diff
@@ -82,6 +84,85 @@ enum MiscSpells
...
+                if (!sSpellMgr->GetSpellInfo(SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL))
+                    return false;
+                return true;
+            }
+
+            void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+            {
+                PreventDefaultAction();
+                // No heal from this talent
+                if (eventInfo.GetHealInfo()->GetSpellInfo()->Id == SPELL_SHAMAN_ANCESTRAL_GUIDANCE_HEAL)
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_shaman.cpp]: use checkproc hook

File: src/server/scripts/Spells/spell_shaman.cpp [hunk 3]
```diff
@@ -82,6 +84,85 @@ enum MiscSpells
...
+                    return;
+                int32 bp0 = CalculatePct(int32(eventInfo.GetDamageInfo()->GetDamage() + eventInfo.GetHealInfo()->GetHeal()), aurEff->GetAmount());
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_shaman.cpp]: this won't work correctly with the current "hacked" implementation of eventinfo, `eventInfo.GetDamageInfo()->GetDamage()` and `eventInfo.GetHealInfo()->GetHeal()` holds the same data

**Comment by Ulduar** [comment]
I know that a talent is not the case, just have a similar spell that does not work either. Thanks for example. I know about prepatch changes. Can suggested only WoD version.
On future i can suggested spell script for death knight Sindragosa Breath Mark.
enum SindragosaBreathMark
{
    SPELL_DK_SINDRAGOSA_BREATH                  = 155168
};
``` CPP
// 155166 - Sindragosa Mark
class spell_dk_sindragosa_breath_2 : public SpellScriptLoader
{
    public:
        spell_dk_sindragosa_breath_2() : SpellScriptLoader("spell_dk_sindragosa_breath_2") { }

        class spell_dk_sindragosa_breath_2_AuraScript : public AuraScript
        {
            PrepareAuraScript(spell_dk_sindragosa_breath_2_AuraScript);

            bool Validate(SpellInfo const* /*spellInfo*/) override
            {
                if (!sSpellMgr->GetSpellInfo(SPELL_DK_SINDRAGOSA_BREATH))
                    return false;
                return true;
            }

            bool CheckProc(ProcEventInfo& eventInfo)
            {
                if (eventInfo.GetHealInfo()->GetSpellInfo()->Id == SPELL_DK_SINDRAGOSA_BREATH)
                    return false;
                return true;
            }

            void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
            {
                PreventDefaultAction();
                int32 bp0 = CalculatePct(int32(eventInfo.GetDamageInfo()->GetDamage()), aurEff->GetAmount());
                if (bp0)
                    eventInfo.GetActionTarget()->CastCustomSpell(SPELL_DK_SINDRAGOSA_BREATH, SPELLVALUE_BASE_POINT0, bp0, eventInfo.GetActionTarget(), true, NULL, aurEff);
            }

            void Register() override
            {
                DoCheckProc += AuraCheckProcFn(spell_dk_sindragosa_breath_2_AuraScript::CheckProc);
                OnEffectProc += AuraEffectProcFn(spell_dk_sindragosa_breath_2_AuraScript::HandleEffectProc, EFFECT_0, SPELL_AURA_DUMMY);
            }
        };

        AuraScript* GetAuraScript() const override
        {
            return new spell_dk_sindragosa_breath_2_AuraScript();
        }
};
```

**Comment by Aokromes** [comment]
@Shauren @joschiwald this can be merged?

**Pull request closed without merging**